### ISSUE TITLE
disallow-reactive-vars-on-mount

### ISSUE THREAD

**Issue by baseballyama** [sveltejs/eslint-plugin-svelte#1205]
### Motivation

onMount runs in the component lifecycle, while reactive variables operate within the reactivity lifecycle. Referencing reactive variables inside the component lifecycle can easily lead to bugs.

### Description

You cannot access reactive variables inside the onMount function.

### Examples

```svelte
<script>
	import { onMount } from 'svelte';
	let count = $state(0);

	// ✗ BAD
	onMount(() => {
		console.log(count);
	});
</script>

<!-- ✓ GOOD -->
{#if count > 0}
	<p>Count is positive</p>
{/if}
```
